playbooks/ORCHESTRATION.md

```diff
@@ -0,0 +1,219 @@
+
+Second, online keys on GCP are used for snapshot, timestamp, and delegation roles. These are defined by a [go-cloud](https://gocloud.dev) style URI to refer to the specific provider like `gcpkms://`. See cosign [KMS integrations](https://github.com/sigstore/cosign/blob/main/KMS.md) for details. These are configured through environment variables in the `scripts/` which propogate to command line flags to the binary.
+
+## Step 0: Building the binary
+
+Run the following script to build the TUF repository binary. 
+```bash
+./scripts/step-0.sh
+```
+
+## Step 1: Root Key Updates
+
+Like mentioned in [Key configuration](#key-configuration), each root key corresponds to a subfolder named by its serial number. The [initialization](#step-2-initializing-a-root-and-targets) script automatically picks up any new subfolders and adds them to the root keys. Any subfolders that are removed are revoked from the root.
+
+### Adding a Root Key
+
+Ask the new root keyholder to run:
+```bash
+export GITHUB_USER=${GITHUB_USER}
+./scripts/step-0.sh && ./scripts/step-1.sh
+```
+
+and verify their PRs with the PR number as the argument to the script:
+
+```bash
+./scripts/verify.sh $PR
+```
+
+You should expect to see their serial number key verified, which should match the committed subfolder.
+
+### Revoking a Root Key
+
+Remove the subfolder. This is done through the [initialization](#step-2-initializing-a-root-and-targets) script by passing the serial number as script argument. See below.
+
+## Step 2: Initializing a root and targets
+
+This step initializes or stages a new root and targets file according to the pre-work and configuration.  Any new key additions from [Step 1](#step-1-root-key-updates) will be picked up. If a key needs to be removed, pass it in as a parameter like follows:
+
+```bash
+./scripts/step-1.5.sh 123456
+```
+This copies over old repository metadata and keys from the `${PREV_REPO}`, revokes key `123456`, and then updates a new root and targets according to the configuration. The new PR will create a new `root.json`, `targets.json`, and delegation files in the `${REPO}/staged` subfolder. You should see the following directory structure created:
+```
+$REPO
+├── keys
+├── repository
+└── staged
+    ├── root.json
+    ├── targets
+    │   └── $TARGET
+    ├── targets.json
+    ├── staging.json
+    ├── rekor.json   
+    └── revocation.json 
+```
+
+Manually check for:
+* The expected root and targets expirations.
+* The expected root and targets versions.
+* The expected keyholders and placeholder signatures.
+* The expected target and delegation files. Check the termination, paths, and targets on each delegation.
+
+<!-- TODO: Add playbook for disaster/recovery steps. -->
+
+### Hardware Key Signing
+
+Next, the root and targets file must be signed. Ask each root keyholder to run:
+
+```bash
+./scripts/step-0.sh && ./scripts/step-2.sh
```

Comment:
Is step-0 always necessary, or is it just so we know the state of the local repo?

Reply:
It's always necessary to build the repo. I added it here because I assume that the orchestrator has run step 0, but the the root keyholder may not have.